### PR TITLE
Lift Microsoft.NETCore.Targets to reduce v1 conflicts

### LINKED ISSUE #18088: Downgrade warnings when referencing log4net in netcoreapp2.0
Copied from https://github.com/dotnet/corefx/issues/17278.

log4net referenced a mix of 1.0 and 1.1 packages.

When used in netcoreapp1.1 this didn't matter, since the netcore.app meta-package would lift all the packages up to 1.1.

In netcoreapp2.0 this no longer happens due to our smaller package graph.

The result is that the mix of 1.0 and 1.1 cause downgrade warnings when 1.1 package bring in the 1.1 lineup (Microsoft.NETCore.Targets).  This is because the lineup brings in 1.1 versions for **all** split packages, including those 1.0 references from log4net.  The 1.1 runtime packages reference 1.1 dependencies which ends up looking like a downgrade to nuget.

We could fix this by shipping a new version of Microsoft.NETCore.Targets in netcoreapp2.0 which has no lineup info in its runtime.json.  That would have the added benefit of reducing the download for mixed 1.x/2.0 projects, reducing the package graph, and thus our reliance on conflict resolution.  The downside is that it would grow the package graph slightly.

@weshaggard @terrajobst what do you think?

### PR DESCRIPTION
**Pull request by ericstj**
Include the update to v1 lineup package, which now has no runtime package
mappings.

This will reduce the number of packages downloaded when mixing v1 and
later packages as well as reduce our dependence on conflict resolution
for dropping those v1 runtime packages.

Fixes #18088

Requires https://github.com/dotnet/core-setup/pull/2053

/cc @weshaggard @joperezr

### CODE REVIEW

**Comment by ericstj** [comment]
@dotnet-bot test this please

File: pkg/Microsoft.NETCore.Targets/runtime.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
+{
+    "supports": {
```
Comment (weshaggard) [inline pkg/Microsoft.NETCore.Targets/runtime.json]: Is there any reason to keep these supports clauses?

File: pkg/frameworkPackage.targets [hunk 2]
```diff
@@ -43,6 +43,15 @@
       <Version>$(PlatformPackageVersion)</Version>
       <TargetFramework>$(TargetFramework)</TargetFramework>
     </Dependency>
+
+    <!-- Include the update to v1 lineup package, which now has no runtime package mappings.
+         This will reduce the number of packages downloaded when mixing v1 and later
+         packages as well as reduce our dependence on conflict resolution for dropping
+         those v1 runtime packages -->
+    <Dependency Condition="'$(IsLineupPackage)' == 'true'" Include="Microsoft.NETCore.Targets">
+      <Version>2.0.0</Version>
```
Comment (weshaggard) [inline pkg/frameworkPackage.targets]: Any reason to not be a ProjectReference as opposed to hardcoding the version? Also is there much reason to reference this in the private transport packages?

Comment (ericstj) [inline pkg/frameworkPackage.targets]: > reason to not be a ProjectReference
It was what we were doing for platforms and it reduces a bit of build time.  I thought about this and another reason was that I didn't expect us to ever update this dependency.  In fact, we'll likely delete the package authoring after 2.0.

> reason to reference this in the private transport packages
I'm just using that to flow the dependency, we could instead add the reference to the package in core-setup if we wanted.

File: pkg/Microsoft.NETCore.Targets/runtime.json (hunk 1 above)
Comment (ericstj) [inline pkg/Microsoft.NETCore.Targets/runtime.json]: Folks who use them would be broken if we removed them.  I felt like removing them was an unnecessary breaking change.

**Comment by ericstj** [comment]
Merging. OSX build seems to be hitting lab issues again.

**Pull request merged**